ecl/hql/hqlrepository.cpp

```diff
@@ -379,21 +379,30 @@ class HQL_API CNewEclRepository : implements IEclRepositoryCallback, public CInt
     virtual IHqlExpression * loadSymbol(IHqlRemoteScope *scope, IIdAtom * searchName) override;
     virtual IEclSource * getSource(IEclSource * parent, IIdAtom * searchName) override;
 
+
 protected:
+    virtual void resolveCollection() { }
+
     IHqlExpression * createSymbol(IHqlRemoteScope * rScope, IEclSource * source);
+    void ensureCollection()
```

Comment:
yes it would be simpler.  The reason I coded it this was was to avoid a virtual call if it had already been set up - so this code is more efficient.  However... this isn't really a case where efficiency is that critical, and it would be clearer, so I may  change as you suggest.